kong/plugins/jwt/jwt_parser.lua

```diff
@@ -282,6 +282,23 @@ function _M:verify_registered_claims(claims_to_verify)
   return errors == nil, errors
 end
 
+--- check the maximum allowed expiration 
+-- @param maximum_expiration of the claim
+-- @return A Boolean indicating true, if the claim the maximum allowed expiration
+-- @return error if any
+function _M:check_maximum_expiration(maximum_expiration)
+  if maximum_expiration <= 0 then
+     return true, nil
+  end
+
+  local exp = self.claims["exp"]
+  if exp == nil or (exp - ngx_time()) > maximum_expiration then
+      return false, {exp = "exceeds maximum allowed expiration"}
```

Comment:
style: the indentation in this codebase is 2 spaces (this one uses 4 and the above branch uses 3)